sat/cli/bootprep/input/image.py

```diff
@@ -644,24 +646,38 @@ def filter_wildcard(self):
         """str or None: the filter pattern to use for filtering images or recipes from the product"""
         return get_val_by_path(self.image_data, 'base.product.filter.wildcard')
 
-    def filter_func(self, name):
-        """Filter the given name.
+    @property
+    @jinja_rendered
+    def filter_arch(self):
+        """str or None: the architecture to use for filtering images or recipes from the product"""
+        return get_val_by_path(self.image_data, 'base.product.filter.arch')
+
+    @property
+    def has_filter(self):
+        """bool: True if a filter is applied, false otherwise"""
+        return (self.filter_prefix is not None or
+                self.filter_wildcard is not None or
+                self.filter_arch is not None)
+
+    def filter_func(self, ims_resource):
+        """Filter the given IMS resource.
+
+        The bootprep input file schema allows multiple filters to be present,
+        in which case all filters must match.
 
         Args:
-            name (str): the name to check against the filter.
+            ims_resource (dict): the IMS resource (image or recipe) to filter
 
         Returns:
-            True if the filter matches the given name or there is no filter, False otherwise
+            True if the filter matches the given IMS resource or there is no filter, False otherwise
         """
-        # Schema validation guarantees that there cannot be both a prefix and a wildcard,
-        # so it is possible to just check each individually and not worry about the case
-        # where both are present.
-        if self.filter_prefix:
-            return name.startswith(self.filter_prefix)
-        if self.filter_wildcard is not None:
-            return fnmatch.fnmatch(name, self.filter_wildcard)
-
-        return True
+        ims_resource_name = ims_resource.get('name', '')
+        return all([
+            not self.filter_prefix or ims_resource_name.startswith(self.filter_prefix),
+            self.filter_wildcard is None or fnmatch.fnmatch(ims_resource_name, self.filter_wildcard),
+            # IMS images or recipes without the 'arch' field are assumed to be x86_64
+            self.filter_arch is None or ims_resource.get('arch', 'x86_64') == self.filter_arch
+        ])
```

Comment:
Perhaps I'm not following. It isn't a list comprehension, but rather a hard-coded list or three different conditions. I thought about whether it could be abstracted and implemented more uniformly across the different filter values, but since the `ims_resource` needs to be checked against each filter value differently, I ended up just doing it this way.

Reply:
Oh, yeah I did misread this. This should be fine then.